src/command-helpers/network.js

```diff
@@ -0,0 +1,135 @@
+  await withSpinner(
+    `Update sources network`,
+    `Failed to update sources network`,
+    `Warnings while updating sources network`,
+    async spinner => {
+      let allNetworks
+
+      try {
+        step(spinner, `Reading networks config`)
+        allNetworks = await toolbox.filesystem.read(networksFile, "json")
+      }  catch (error) {
+        throw new Error(error.message)
+      }
+
+      let networkConfig = allNetworks[network]
+
+      // Exit if the network passed with --network does not exits in networks.json
+      if(!networkConfig) {
+        throw new Error(`Network '${network}' was not found in '${networksFile}'`)
+      }
+
+      await toolbox.patching.update(manifest, content => {
+        let subgraph = yaml.parse(content)
+        let networkSources = Object.keys(networkConfig)
+        let subgraphSources = subgraph.dataSources.map(value => value.name);
+
+        // Update the dataSources network config
+        subgraph.dataSources = updateSources(
+          toolbox,
+          spinner,
+          network,
+          subgraph.dataSources,
+          networkSources,
+          networkConfig
+        )
+
+        // All data sources shoud be on the same network,
+        // so we have to update the network of all templates too.
+        if(subgraph.templates) {
+          subgraph.templates = updateTemplates(network, subgraph.templates)
+        }
+
+        let unsusedSources = networkSources.filter(x => !subgraphSources.includes(x))
+
+        unsusedSources.forEach(source => {
+          step(spinner, `dataSource '${source}' from '${networksFile}' not found in ${manifest}`)
+        })
+
+        let yaml_doc = new yaml.Document()
+        yaml_doc.contents = subgraph
+        return yaml_doc.toString()
+      })
+  })
+
+const initNetworksConfig = async(toolbox, directory) =>
+  await withSpinner(
+    `Initialize networks config`,
+    `Failed to initialize networks config`,
+    `Warnings while initializing networks config`,
+    async spinner => {
+      let subgraphStr = await toolbox.filesystem.read(path.join(directory, 'subgraph.yaml'))
+      let subgraph = yaml.parse(subgraphStr)
+
+      let networks = {}
+
+      subgraph.dataSources.forEach(source => {
+        let sourceConfig = {[source.name]: {}}
+
+        if (source.source.address) sourceConfig[source.name]["address"] = source.source.address
+        if (source.source.startBlock) sourceConfig[source.name]["startBlock"] = source.source.startBlock
+
+        networks[source.network] = sourceConfig
+      })
+
+      await toolbox.filesystem.write(`${directory}/networks.json`, networks)
+
+      return true
+    },
+  )
+
+// Iterates through each dataSource and updates the network configuration
+// Will update only if there are changes to the configuration
+function updateSources(toolbox, spinner, network, sources, networkSources, networkConfig) {
+  sources.forEach(source => {
```

Comment:
Same thing about `updateTemplates`, I would do `.map` instead of `.forEach` since you're already reassigning the value.